src/pages/index.js

```diff
@@ -63,48 +100,158 @@ Array.from(document.forms).forEach((formElement) => {
 });
 /* END DECLARATIVE SECTION */
 
-/* PROFILE SECTION */
+/* PROFILE INFO SECTION */
 const userProfile = new UserInfo({
+  // pass as arguments the HTML DOM profile elements
   name: profileName,
   description: profileDescription,
+  avatar: profileAvatar,
 });
 
-const popupProfile = new PopupWithForm(
+buttonEditProfile.addEventListener("click", () => {
+  const { name, description } = userProfile.getUserInfo();
+  inputProfileName.value = name;
+  inputProfileDescription.value = description;
+  formValidators["profile_info_form"].resetFormValidation(); // input always valid when taking from the html page
+  popupProfileInfo.open();
+});
+
+const popupProfileInfo = new PopupWithForm(
   modalWindowProfile,
   (event, { name, description }) => {
     event.preventDefault();
-    userProfile.setUserInfo(name, description);
-    popupProfile.close();
+    popupProfileInfo.setRender(true); // renders faster when synced before the async begins
+    api
+      .setUserProfileData(name, description)
+      .then(() => {
+        userProfile.setUserInfo(name, description);
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        popupProfileInfo.close();
```

Comment:
`        popupProfileInfo.close();
`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) **Please, correct it everywhere in the code**